### ISSUE TITLE
Incorrect Retrograde Factor Calculation

### ISSUE THREAD

**Issue by thkruz** [david-rc-dayton/pious_squid#1]
I am working on expanding the documentation and testing on my TypeScript port and this looks like an error:

https://github.com/david-rc-dayton/pious_squid/blob/d3579648da9be426c76c910e4c12c460171e86bd/lib/src/coordinate/classical_elements.dart#L170C5-L170C59

Based on the variables you chose to use, I am guessing you are basing it in Vallado's work. I found a good reference to his code in Matlab here https://github.com/Spacecraft-Code/Vallado/blob/e5206649ae7004ba11823b444d7fd53a82a264e2/Matlab/rv2eq.m#L52C1-L54C13

Along with a great [writeup from NPS](https://faculty.nps.edu/dad/orbital/th0.pdf) - see section 2.1.2

In both cases it looks like this should be a basic check if inclination is greater than PI/2, which feels intuitive.

Am I missing something or is this a mistake?

**Comment by david-rc-dayton** [comment]
@thkruz Good catch, the current code does seem squiffy. I'll do some digging.

**Comment by david-rc-dayton** [comment]
Looks like the original reference was the C# Vallado code that has the weird retrograde check: https://github.com/Posnet/Vallado/blob/7bed35532e142e5f959e28d245c5ecd773ffbbc4/cs/AstroLib/Astrolib/AstroLib.cs#L4076-L4077

The C++ code seems to match the Matlab implementation for the retrograde factor, so I'm going to use that in the next release: https://github.com/Posnet/Vallado/blob/master/cpp/astrolib/AstroLib/AstroLib.cpp#L3449-L3450

```typescript
function radians(deg: number): number {
  return deg * (Math.PI / 180);
}

function frMatlab(inclination: number): -1 | 1 {
  return inclination > Math.PI * 0.5 ? -1 : 1;
}

function frCSharp(inclination: number): -1 | 1 {
  return Math.abs(inclination - Math.PI) < 1e-4 ? -1 : 1;
}

for (let deg = 0; deg <= 180; deg += 5) {
  const rad = radians(deg);
  const matlab = frMatlab(rad);
  const csharp = frCSharp(rad);
  const flag = matlab === csharp ? "" : "****";
  console.log(`i=${deg}\u00b0 matlab=${matlab} csharp=${csharp} ${flag}`);
}
// i=0°   matlab=1  csharp=1
// i=5°   matlab=1  csharp=1
// i=10°  matlab=1  csharp=1
// i=15°  matlab=1  csharp=1
// i=20°  matlab=1  csharp=1
// i=25°  matlab=1  csharp=1
// i=30°  matlab=1  csharp=1
// i=35°  matlab=1  csharp=1
// i=40°  matlab=1  csharp=1
// i=45°  matlab=1  csharp=1
// i=50°  matlab=1  csharp=1
// i=55°  matlab=1  csharp=1
// i=60°  matlab=1  csharp=1
// i=65°  matlab=1  csharp=1
// i=70°  matlab=1  csharp=1
// i=75°  matlab=1  csharp=1
// i=80°  matlab=1  csharp=1
// i=85°  matlab=1  csharp=1
// i=90°  matlab=1  csharp=1
// i=95°  matlab=-1 csharp=1  ****
// i=100° matlab=-1 csharp=1  ****
// i=105° matlab=-1 csharp=1  ****
// i=110° matlab=-1 csharp=1  ****
// i=115° matlab=-1 csharp=1  ****
// i=120° matlab=-1 csharp=1  ****
// i=125° matlab=-1 csharp=1  ****
// i=130° matlab=-1 csharp=1  ****
// i=135° matlab=-1 csharp=1  ****
// i=140° matlab=-1 csharp=1  ****
// i=145° matlab=-1 csharp=1  ****
// i=150° matlab=-1 csharp=1  ****
// i=155° matlab=-1 csharp=1  ****
// i=160° matlab=-1 csharp=1  ****
// i=165° matlab=-1 csharp=1  ****
// i=170° matlab=-1 csharp=1  ****
// i=175° matlab=-1 csharp=1  ****
// i=180° matlab=-1 csharp=-1
```
^ the matlab logic makes more sense

**Comment by david-rc-dayton** [comment]
Fixed in `0.3.0` release.
